docs/modules/sql/pages/mapping-to-kafka.adoc

```diff
@@ -30,27 +30,37 @@ To create a mapping to a Kafka topic in SQL, you must use the `CREATE MAPPING` s
 - The address of the Kafka broker
 - Client authentication details
 - How to serialize/deserialize the keys and values in messages
+- The preferred local parallelism for the connector (for source mappings)
 
 ```sql
 CREATE MAPPING my_topic <1>
 TYPE Kafka <2>
 OPTIONS (
     'bootstrap.servers' = '127.0.0.1:9092', <3>
-    'security.protocol'='SASL_SSL', 
-    'sasl.jaas.config'='org.apache.kafka.common.security.plain.PlainLoginModule
+    'security.protocol' = 'SASL_SSL',
+    'sasl.jaas.config' = 'org.apache.kafka.common.security.plain.PlainLoginModule
     required username="<my_api_key>"
     password="<my_api_secret>";',
-    'sasl.mechanism'='PLAIN', <4>
+    'sasl.mechanism' = 'PLAIN', <4>
+    'preferredLocalParallelism' = '2' <5>
 );
 ```
 
 <1> The name of the Kafka topic.
 <2> The name of the connector.
 <3> The address of the Kafka broker.
 <4> The SASL configuration used to authenticate client connections, where the API key provides the username and the API secret key provides the password. Note that the clause must end with a semi-colon, as shown in the example above.
+<5> The preferred local parallelism for the connector (for source mappings). If not specified, the default is `4`. See below for more information.
 
 NOTE: Any key/value pairs in the `OPTIONS()` function that are not recognized by Hazelcast are passed directly to the Kafka consumer/producer.
 
+Each member of the Hazelcast cluster has a processor running a Kafka connector for the mapping.
+For source mappings, the preferred local parallelism is how many parallel processors there are on each member.
```

Comment:
Technically, the PLP number depends on the number partitions in the source Kafka topic, not the implementation of the connector. The connector belongs to Hazelcast, not the user, so it should not be phrased as "your connector".

I think the suggested added sentence is confusing because a user should always think about what PLP to use and not omit it except for initial testing or experimentation.

Reply:
Thank you for the explanation; my suggestion was based on a line in your description in the related code change PR, I have only basic knowledge of the area.

So long as the implication that the specified _preferred_ number might not match the _actual_ number is going to be understood by the users, I'm happy for it to be removed. If, however, there is any chance that someone is surprised at this mismatch, I feel it best to explain it explicitly (even if my attempt was inexact!). I'll leave it to you to decide whether such an explanation is required or not.

In the meantime, I have amended my suggested change and removed "Whether the preferred number is used depends on the implementation of your connector."

Reply:
Thanks for the feedback. Documented this is indeed a bit tricky, not so easy for me either :)
I will commit your suggestions and give it another pass 👍.